experimental/dds/tree2/src/core/forest/forest.ts

```diff
@@ -99,7 +100,16 @@ export interface IForestSubscription extends Dependee, ISubscribable<ForestEvent
 	 * This dummy node can be used to read the detached fields,
 	 * but other operations (such as inspecting the dummy node's type or path) should not be relied upon.
 	 */
-	moveCursorToPath(destination: UpPath | undefined, cursorToMove: ITreeSubscriptionCursor): void;
+	moveCursorToPath(destination: UpPath, cursorToMove: ITreeSubscriptionCursor): void;
+
+	/**
+	 * The cursor is moved to a special dummy node above the detached fields.
+	 * This dummy node can be used to read the detached fields,
+	 * but other operations (such as inspecting the dummy node's type or path) should not be relied upon.
+	 * As this method does not return a forest cursor and cannot be freed, the caller of this method must
+	 * use with caution and ensure that it is not used incorrectly.
+	 */
+	getRootCursor(): ITreeCursorSynchronous;
```

Comment:
You are adding a method to the forest API: the forest test suite should be updated to include tests for it.. Apparently it was missing tests for this case before (which enabled this problem in the first place). It would be good to ensure moveCursorToPath has tests for the other cases (that are still supported) as well.

Reply:
Added unit tests for `moveCursorToPath` and `getCursorAboveDetachedFields`.

I created separate tests for these methods (not as a part of `forestTestSuite.ts`), as the `moveCursorToPath` for the ObjectForest has an additional test case (path set to `undefined`)